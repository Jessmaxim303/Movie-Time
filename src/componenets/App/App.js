import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.scss';
import MovieContainer from '../../containers/MovieContainer/MovieContainer.js';
import Login from '../Login/Login';
import MovieDetail from '../MovieDetail/MovieDetail.js';
import Header from '../Header/Header.js';
import { fetchMoviesAPI } from '../../apiCalls/apiCalls.js';
import { loadingMovies, getMovies } from '../../actions';
import { Loading } from '../Loading/Loading.js';
import PropTypes from 'prop-types';
import Background from '../../Assets/login-backdrop.jpg';

export class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount() {
    Promise.all([this.loadMovies()])
  }

  loadMovies() {
    fetchMoviesAPI()
    .then(data => {
      this.props.getMovies(data.movies)
      this.props.loadingMovies(true)
    })
  }

  render() {
    const backgroundStyling = {
      background: `linear-gradient(to top, rgba(0, 0, 0, .7), rgba(42, 42, 42, 0)), url(${Background}) no-repeat center top`,
      backgroundSize: 'cover',
    }
    if (this.props.loadingStatus) {
      return (
      <main className='main'>
        <Header />
        <Loading />
      </main>
      )
    }
    return (
      <main className='main' >
        
        <div className='content' style={backgroundStyling}>
          <Switch>
            <Route
              path='/movie'
              render={() => <MovieContainer />}
            />
            <Route
              path='/movies/:id'
              render={({ match })=>< MovieDetail selectedMovie=
                  {this.props.movies.find(movie => movie.id === parseInt(match.params.id))}/>}/>
            <Route
              path='/'
              render={()=><Login />} />
          </Switch>
        </div>
      </main>
    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  ratings: state.ratings,
  loadingStatus: state.loadingStatus,
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  loadingMovies: (loadingStatus) => dispatch(loadingMovies(loadingStatus)),
  getMovies: (movies) => dispatch(getMovies(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  movies: PropTypes.array,
  ratings: PropTypes.array,
  loadingStatus: PropTypes.number,
  user: PropTypes.string
}
