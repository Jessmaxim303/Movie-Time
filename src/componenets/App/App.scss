@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';

.main {
  @include display-flex (column, default, default);
  display: flex;
  flex-direction: column;
  font-family: $main-font;
  height: 100vh;
  backgroundImage: url('../../Assets/login-backdrop.jpg');
  /*background: blue;*/
}

.content {
  @include display-flex (column, default, default);
  flex: 1;
}
