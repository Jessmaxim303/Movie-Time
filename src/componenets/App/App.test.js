import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App.js';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with the movies and ratings array and a loading status', () => {
      const mockState = {
        movies: [{title: 'Parasite', rating: 9}, {title: '1917', rating: 8}],
        ratings: [{id: 25, rating: 9}, {id: 27, rating: 4}],
        user: {id: 123, name: 'me'},
        loadingStatus: true
      }
      const expected = {
        movies: [{title: 'Parasite', rating: 9}, {title: '1917', rating: 8}],
        ratings: [{id: 25, rating: 9}, {id: 27, rating: 4}],
        loadingStatus: true
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch with a postRating action when ')
  // })
})
