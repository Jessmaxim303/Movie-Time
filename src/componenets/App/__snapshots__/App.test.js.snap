// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
<main
  className="main"
>
  <Connect(Header) />
  <div
    className="content"
  >
    <Switch>
      <Route
        path="/login"
        render={[Function]}
      />
      <Route
        path="/movies/:id"
        render={[Function]}
      />
      <Route
        path="/"
        render={[Function]}
      />
    </Switch>
  </div>
</main>
`;
